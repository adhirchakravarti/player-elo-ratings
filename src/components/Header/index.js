import React, { useState } from "react";
import {
  AppBar,
  Toolbar,
  IconButton,
  Icon,
  Typography,
  makeStyles,
} from "@material-ui/core";
import SideDrawer from "../SideDrawer";

const useStyles = makeStyles((theme) => ({
  menuButton: {
    marginRight: theme.spacing(2),
  },
}));

function Header() {
  const classes = useStyles();
  const [drawerState, setDrawerState] = useState({
    left: false,
  });

  const toggleDrawer = (anchor, open) => {
    setDrawerState({ ...drawerState, [anchor]: open });
  };

  return (
    <>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            color="inherit"
            aria-label="menu"
            className={classes.menuButton}
            onClick={() => toggleDrawer("left", true)}
          >
            <Icon>menu</Icon>
          </IconButton>
          <Typography variant="h6">Player Ratings</Typography>
        </Toolbar>
      </AppBar>
      <SideDrawer
        open={drawerState["left"]}
        onClose={() => toggleDrawer("left", false)}
      />
    </>
  );
}

export default Header;
