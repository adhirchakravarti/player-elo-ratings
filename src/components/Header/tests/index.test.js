import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../index";

const mockProps = {
  title: "Scopely Player Ratings",
};

describe("<SideDrawer />", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header {...mockProps} />
      </MemoryRouter>
    );
    const initialRender = asFragment();
    expect(initialRender).toMatchSnapshot();
  });
  it("does not log errors in the console", () => {
    const spy = jest.spyOn(global.console, "error");
    render(
      <MemoryRouter>
        <Header {...mockProps} />
      </MemoryRouter>
    );
    expect(spy).not.toHaveBeenCalled();
  });
  it("should contain the title that was sent as a prop", () => {
    const { getByText } = render(
      <MemoryRouter>
        <Header {...mockProps} />
      </MemoryRouter>
    );
    const title = getByText("Scopely Player Ratings").textContent;
    expect(title).toEqual(mockProps.title);
  });
});
