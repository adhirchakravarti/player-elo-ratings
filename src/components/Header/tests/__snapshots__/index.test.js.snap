// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-1 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root"
          >
            menu
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Scopely Player Ratings
      </h6>
    </div>
  </header>
  <div />
</DocumentFragment>
`;
